share/functions/cd.fish

```diff
@@ -4,7 +4,7 @@
 function cd --description "Change directory"
     set -l MAX_DIR_HIST 25
 
-    if test (count $argv) -gt 1
+    if test (count $argv) -gt (test $argv[1] = "--" && echo 2 || echo 1)
```

Comment:
This `$argv[1]` needs to be quoted - in general any variable used with `test` needs to be quoted.

Otherwise this is a good, simple, solution.

Reply:
Thank you for this advice. I added the quotes and the build passed.